Platform/AMD/AmdMinBoardPkg/Library/PeiBoardInitPreMemLib/PeiMemoryInit.c

```diff
@@ -89,7 +89,22 @@ EndofAmdMemoryInfoHobPpiGuidCallBack (
             FixedPcdGet32 (PcdAmdSmramAreaSize)
             ));
 
-          AmdMemoryInfoRange->Size -= FixedPcdGet32 (PcdAmdSmramAreaSize);
+          if (AmdMemoryInfoRange->Size) {

```

Comment:
As ->Size is not a boolean, we should have the predicate expressions.  Such as if (AmdMemoryInfoRange->Size > 0).